File: easy/one_zero_two_zero.rb
File.open(ARGV[0]).each_line do |line|
  s = line.split.map(&:to_i)
  r = 0
  (1 << s[0]).upto(s[1]) do |i|
    n = s[0]
    a = i
    while a > 0 && n >= 0
      n -= 1 if a.even?
      a /= 2
    end
    r += 1 if n == 0
  end
  puts r
end

File: easy/one_zero_two_zero.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    my ( $n, $a ) = ( $line =~ /(\d+) (\d+)/ );
    my $r = 0;
    for ( my $i = 1 << $n ; $i <= $a ; $i++ ) {
        my ( $m, $b ) = ( $n, $i );
        while ( $b > 0 && $m >= 0 ) {
            $m-- if ( $b & 1 ) == 0;
            $b >>= 1;
        }
        $r++ if $m == 0;
    }
    print "$r\n";
}
close(INFILE);
